src/jit/lir.cpp

```diff
@@ -525,10 +468,10 @@ GenTree* LIR::Range::FirstNonPhiNode() const
             continue;
         }
 
-        break;
+        return node;
     }
 
-    return node;
+    return nullptr;
```

Comment:
How about also having:

GenTree* LIR::Range::LastPhiNode() const
{
    GenTree* lastPhi = nullptr;
    for (GenTree* node : *this)
    {
        if (node->OperGet() == GT_PHI_ARG)
        {
            lastPhi = node;
            continue;
        }
        else if (node->OperGet() == GT_PHI)
        {
            lastPhi = node;
            continue;
        }
        else if (node->IsPhiDefn())
        {
            lastPhi = node;
            continue;
        }

        return lastPhi;
    }

    return nullptr;
}